### PR TITLE
[WIP] Add warm_start  parameter to SCML

### PR DESCRIPTION
**Pull request by maxi-marufo**
Because [SCML](http://researchers.lille.inria.fr/abellet/papers/aaai14.pdf) optimization procedure is based on stochastic subgradient descent, we can save the weights after fitting the model, and use them in a following fit call (with a different set of triplets). The decision to use the `warm_start` parameter instead of a new `partial_fit` method is because [partial_fit in scikit-learn will only fit 1 epoch, whereas fit will fit for multiple epochs (until the loss converges or max_iter is reached)](https://stackoverflow.com/questions/38052342/what-is-the-difference-between-partial-fit-and-warm-start), which is the case also for SCML.

### CODE REVIEW

**Comment by perimosocordiae** [comment]
This change looks fine to me, though I'm not sure when this warm-start option is useful in practice. Sorry for the extreme delay in reviewing!

@grudloff want to take a look?

**Comment by grudloff** [comment]
LGTM!

**Review by bellet — COMMENTED** [review]
Thanks @maxi-marufo !
Maybe it is possible to design a better test, something in the spirit https://github.com/scikit-learn/scikit-learn/blob/80598905e517759b4696c74ecc35c6e2eb508cff/sklearn/linear_model/tests/test_sgd.py#L274
where the idea is to check that calling `warm_start=True` is equivalent to manually setting the parameters?

**Review by bellet — APPROVED** [review]
LGTM @maxi-marufo, but I would like to rerun the CIs, not sure how to do this - @perimosocordiae @terrytangyuan do you know?

One simple way sould be for you @maxi-marufo to update your branch to the current state of master and push an empty commit

**Comment by grudloff** [comment]
@bellet For a manual run you need to have a trigger on `workflow_dispatch ` but currently it is only set up with push and pull requests to master. So it would not be possible rn
(see https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow)

**Comment by terrytangyuan** [comment]
This just needs a rebase to trigger the CI.

**Comment by terrytangyuan** [comment]
CI triggered

**Comment by bellet** [comment]
@maxi-marufo @terrytangyuan well looks like checks have been cancelled and/or failed?

**Comment by terrytangyuan** [comment]
I think some failed which triggered cancellation of the rest.
